client/changelist/file_changelist.go

```diff
@@ -94,6 +115,41 @@ func (cl FileChangelist) Close() error {
 	return nil
 }
 
+// NewIterator creates an iterator from FileChangelist
+func (cl FileChangelist) NewIterator() (ChangeIterator, error) {
+	fileInfos, err := getFileNames(cl.dir)
+	if err != nil {
+		return &FileChangeListIterator{}, err
+	}
+	sort.Sort(fileChanges(fileInfos))
+	return &FileChangeListIterator{dirname: cl.dir, collection: fileInfos}, nil
+}
+
+// FileChangeListIterator is a concrete instance of ChangeIterator
+type FileChangeListIterator struct {
+	index      int
+	dirname    string
+	collection []os.FileInfo
+}
+
+// Next returns the next Change in the FileChangeList
+func (m *FileChangeListIterator) Next() (item Change, err error) {
+	defer func() {
+		if e := recover(); e != nil {
+			err = fmt.Errorf("Error in Next(): %v ", e)
+		}
+	}()
+	f := m.collection[m.index]
+	m.index++
+	item, err = unmarshalFile(m.dirname, f)
+	return
+}
+
+// HasNext indicates whether iterator is exhausted
+func (m *FileChangeListIterator) HasNext() bool {
+	return m.index < len(m.collection)
+}
```

Comment:
Ah you're right. I should just defer to you :-P

@davidwilliamson at the point where Go actually reads the file, the Go devs already put in this defer...recover step: http://golang.org/src/io/ioutil/ioutil.go#L23